apps/backend/src/common/guards/auth.guard.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  ForbiddenException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import axios, { AxiosInstance } from 'axios';
+import { Request } from 'express';
+import jwt from 'jsonwebtoken';
+import { PUBLIC_ENDPOINT } from './../decorators/publicEndpoint.decorator';
+
+let axiosInstance: AxiosInstance;
```

Comment:
Curious why we are defining axios instance outside of the class but assigning the value in the constructor?